persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/SampleSparkRepository.java

```diff
@@ -0,0 +1,132 @@
+                                                      String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaSamplesOfPatientInStudy(String studyId, String patientId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> getAllSamplesOfPatientsInStudy(String studyId, List<String> patientIds, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> getSamplesOfPatientsInMultipleStudies(List<String> studyIds, List<String> patientIds,
+                                                              String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> fetchSamples(List<String> studyIds, List<String> sampleIds, String projection) {
+
+        Dataset<Row> samples = parquetLoader.loadStudyFiles(spark,
+            new HashSet<>(studyIds), ParquetConstants.CLINICAL_SAMPLE, true);
+        
+        if (!CollectionUtils.isEmpty(sampleIds)) {
+            samples = samples.where(samples.col("SAMPLE_ID").isin(sampleIds.toArray()));
+        }
+        
+        samples.createOrReplaceTempView("samples");
+        StringBuilder sb = new StringBuilder("SELECT PATIENT_ID, SAMPLE_ID, studyId");
+        if (PersistenceConstants.SUMMARY_PROJECTION.equalsIgnoreCase(projection)
+            || PersistenceConstants.DETAILED_PROJECTION.equalsIgnoreCase(projection)) {
+            sb.append(", SAMPLE_TYPE");
+        }
+        sb.append(" FROM samples");
+
+        Dataset<Row> res = spark.sql(sb.toString());
+        
+        return res.collectAsList().stream().
+            map(r -> mapToSample(r, projection)).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<Sample> fetchSamples(List<String> sampleListIds, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaSamples(List<String> studyIds, List<String> sampleIds) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaSamples(List<String> sampleListIds) {
+        throw new UnsupportedOperationException();    }
+
+    @Override
+    public List<Sample> getSamplesByInternalIds(List<Integer> internalIds) {
+        throw new UnsupportedOperationException();
+    }
+    
+    private Sample mapToSample(Row row, String projection) {
+        Sample sample = new Sample();
+        sample.setPatientStableId(row.getString(0));
+        sample.setStableId(row.getString(1));
+        sample.setCancerStudyIdentifier(row.getString(2));
+        if (PersistenceConstants.SUMMARY_PROJECTION.equalsIgnoreCase(projection)
+            || PersistenceConstants.DETAILED_PROJECTION.equalsIgnoreCase(projection)) {
+            Sample.SampleType sampleType = SampleTypeUtil.getType(sample.getStableId(), row.getString(3));
```

Comment:
The SampleTypeUtil borrows from [StableIdUtil](https://github.com/cBioPortal/cbioportal/blob/master/core/src/main/java/org/mskcc/cbio/portal/util/StableIdUtil.java) which is TCGA specific.  Did you consider using the [SampleType](https://github.com/cBioPortal/cbioportal/blob/master/model/src/main/java/org/cbioportal/model/Sample.java#L7) enum defined in the Sample Class?

Reply:
Do you mean adding the util method in the Sample class?

Reply:
I was thinking of moving the SampleTypeTypeHandler out of the mybatis package and generalizing it, but I see that it uses specific Mybatis types.  I do see you are using Sample.SampleType in your util class, which is the most important. Let just leave the way it is.